browser/ui/webui/settings/brave_adblock_handler.cc

```diff
@@ -253,15 +284,68 @@ void BraveAdBlockHandler::ViewSubscriptionSource(
 }
 
 void BraveAdBlockHandler::UpdateCustomFilters(const base::Value::List& args) {
-  if (!args[0].is_string())
+  if (!args[0].is_string()) {
     return;
+  }
 
   std::string custom_filters = args[0].GetString();
   g_brave_browser_process->ad_block_service()
       ->custom_filters_provider()
       ->UpdateCustomFilters(custom_filters);
 }
 
+void BraveAdBlockHandler::GetCustomScriptlets(const base::Value::List& args) {
+  CHECK(base::FeatureList::IsEnabled(
+      brave_shields::features::kCosmeticFilteringCustomScriptlets));
+  CHECK(args.size() == 1u && args[0].is_string());
+  AllowJavascript();
+
+  const auto& custom_resources = g_brave_browser_process->ad_block_service()
+                                     ->custom_resource_provider()
+                                     ->GetCustomResources();
+
+  ResolveJavascriptCallback(args[0].GetString(), custom_resources);
+}
+
+void BraveAdBlockHandler::AddCustomScriptlet(const base::Value::List& args) {
+  CHECK(base::FeatureList::IsEnabled(
+      brave_shields::features::kCosmeticFilteringCustomScriptlets));
+  CHECK(args.size() == 2u && args[0].is_string() && args[1].is_dict());
+  AllowJavascript();
+  auto error_code = g_brave_browser_process->ad_block_service()
+                        ->custom_resource_provider()
+                        ->AddResource(args[1]);
+  ResolveJavascriptCallback(args[0].GetString(),
+                            base::Value(static_cast<int>(error_code)));
+}
+
+void BraveAdBlockHandler::UpdateCustomScriptlet(const base::Value::List& args) {
+  CHECK(base::FeatureList::IsEnabled(
+      brave_shields::features::kCosmeticFilteringCustomScriptlets));
+  CHECK(args.size() == 3u && args[0].is_string() && args[1].is_string() &&
+        args[2].is_dict());
+  AllowJavascript();
+  auto error_code = g_brave_browser_process->ad_block_service()
+                        ->custom_resource_provider()
+                        ->UpdateResource(args[1].GetString(), args[2]);
+  ResolveJavascriptCallback(args[0].GetString(),
+                            base::Value(static_cast<int>(error_code)));
+}
+
+void BraveAdBlockHandler::RemoveCustomScriptlet(const base::Value::List& args) {
+  CHECK(base::FeatureList::IsEnabled(
+      brave_shields::features::kCosmeticFilteringCustomScriptlets));
+  AllowJavascript();
+  CHECK(args.size() == 2u && args[0].is_string() && args[1].is_string());
+  AllowJavascript();
```

Comment:
why does AllowJavascript have to be called twice here?

Reply:
Nice catch, needs to be removed